hack/common.mk

```diff
@@ -11,22 +11,24 @@ endif
 tests = $(shell hack/run_test --list 2>/dev/null)
 
 script_env = \
+	TAG_ON_SUCCESS=$(TAG_ON_SUCCESS)                \
 	TEST_CASE="$(TEST_CASE)"                        \
 	SKIP_SQUASH=$(SKIP_SQUASH)                      \
 	UPDATE_BASE=$(UPDATE_BASE)                      \
-	VERSIONS="$(VERSIONS)"                          \
 	OS=$(OS)                                        \
-	VERSION="$(VERSION)"                            \
 	BASE_IMAGE_NAME=$(BASE_IMAGE_NAME)              \
 	OPENSHIFT_NAMESPACES="$(OPENSHIFT_NAMESPACES)"
 
 .PHONY: build
-build:
-	$(script_env) $(build)
+build: $(VERSIONS)
+
+.PHONY: $(VERSIONS)
+$(VERSIONS):
+	VERSION=$@ TEST_MODE=$(TEST_MODE) $(script_env) $(build)
```

Comment:
Haven't run into any issues when running this (together with #149) in parallel yet. Is there something you know of that could be a problem for parallel runs?
The only thing that comes to my mind at this moment is that running Docker builds in parallel causes Docker to pull common ancestors (rhel/centos images) at the same time. Will have to check if there are no hidden issues with that.